pkg/skaffold/build/cluster/cluster.go

```diff
@@ -42,7 +47,47 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, tags tag.ImageTags,
 		defer teardownDockerConfigSecret()
 	}
 
-	return build.InParallel(ctx, out, tags, artifacts, b.buildArtifactWithKaniko)
+	kanikoArtifacts, otherArtifacts := splitArtifacts(artifacts)
+
+	pArts, parallelErr := build.InParallel(ctx, out, tags, kanikoArtifacts, b.buildArtifactWithKaniko)
+	if parallelErr != nil {
+		logrus.Warnf("error building kaniko artifacts in parallel: %v", parallelErr)
+	}
+
+	sArts, err := build.InSequence(ctx, out, tags, otherArtifacts, b.runBuildForArtifact)
+	if err != nil {
+		return append(pArts, sArts...), err
+	}
+
+	return append(pArts, sArts...), parallelErr
+}
+
+func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, artifact *latest.Artifact, tag string) (string, error) {
+	switch {
+	case artifact.KanikoArtifact != nil:
+		return b.buildArtifactWithKaniko(ctx, out, artifact, tag)
+
+	case artifact.CustomArtifact != nil:
+		return b.buildArtifactWithCustomBuilder(ctx, out, artifact, tag)
+
+	default:
+		return "", fmt.Errorf("undefined artifact type: %+v", artifact.ArtifactType)
+	}
+}
+
+func (b *Builder) buildArtifactWithCustomBuilder(ctx context.Context, out io.Writer, artifact *latest.Artifact, tag string) (string, error) {
```

Comment:
one last thing: can we cover that the right env vars are passed in to the builder in tests?

Reply:
done